src/htm/algorithms/SDRClassifier.hpp

```diff
@@ -242,7 +242,7 @@ class Predictor : public Serializable
    *
    * @returns: A mapping from prediction step to PDF.
    */
-  Predictions infer(UInt recordNum, const SDR &pattern);
+  Predictions infer(UInt recordNum, const SDR &pattern); //TODO should recordNum be optional? I think we need only SDR to learn/make predictions
```

Comment:
@ctrl-z-9000-times is there any reason to have `recordNum` required? For both Classifier, Predictor. Would make sense to me to call with just a SDR. Related to #667

Reply:
The `recordNum` was needed when infer added the given data to the classifier's history.  Now that infer has changed, the record num argument could be removed (or made optional and deprecated).

Reply:
> , the record num argument could be removed (or made optional and deprecated).

I'll make the change, probably rm it right away. 

OT: Could Classifier be improved by optional anomalyScore? We'd ignore anomalous SDRs from learning, or otherwise incorporate the information..